homeassistant/components/ecovacs/config_flow.py

```diff
@@ -77,11 +97,22 @@ async def async_step_user(
                                 type=selector.TextSelectorType.PASSWORD
                             )
                         ),
-                        vol.Required(CONF_COUNTRY): vol.All(vol.Lower, cv.string),
-                        vol.Required(CONF_CONTINENT): vol.All(vol.Lower, cv.string),
+                        vol.Required(CONF_COUNTRY): selector.CountrySelector(),
+                        vol.Optional(CONF_CONTINENT): selector.SelectSelector(
+                            selector.SelectSelectorConfig(
+                                options=["as", "eu", "na", "ww"],
+                                translation_key=CONF_CONTINENT,
+                                custom_value=True,
```

Comment:
Why do we allow custom values. I don't expect a new continent to arise in the nearby future.

(Or if you are going to create your own, let me know, I'm interested)

Reply:
The problem is that I have no way to identify all continents used by ecovacs as they use `CNAME` DNS entries, which can not queried to identify it.

Previously, this field was a normal string to which users could add different continents as the above ones.
We could also allow only the above ones, and if another continent is used, ask the user to create an issue.

Reply:
Removed continent completely